### PR TITLE
Add bztar support to meson dist

### PR DESCRIPTION
**Pull request by tristan957**
Some projects, like Postgres, distribute code in this format.

### CODE REVIEW

File: mesonbuild/mdist.py [hunk 1]
```diff
@@ -32,9 +32,10 @@
     from ._typing import ImmutableListProtocol
     from .mesonlib import ExecutableSerialisation
 
-archive_choices = ['gztar', 'xztar', 'zip']
+archive_choices = ['bztar', 'gztar', 'xztar', 'zip']
```
Comment (bruchar1) [inline mesonbuild/mdist.py:35]: nitpick: This list is only used in `determine_archives_to_generate` function, and could easily be replaced by using the keys of `archive_extension`. I would rather remove it...

Comment (tristan957) [reply mesonbuild/mdist.py:35]: True. If someone else comments the same, I will remove it.

Comment (eli-schwartz) [reply mesonbuild/mdist.py:35]: I think it's fine as is.

File: mesonbuild/mdist.py [hunk 2]
```diff
@@ -46,7 +47,7 @@ def add_arguments(parser: argparse.ArgumentParser) -> None:
     parser.add_argument('--allow-dirty', action='store_true',
                         help='Allow even when repository contains uncommitted changes.')
     parser.add_argument('--formats', default='xztar',
-                        help='Comma separated list of archive types to create. Supports xztar (default), gztar, and zip.')
+                        help='Comma separated list of archive types to create. Supports xztar (default), bztar, gztar, and zip')
```
Comment (eli-schwartz) [inline mesonbuild/mdist.py:50]: Why is the period being deleted here?

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: mesonbuild/mdist.py (hunk 2 above)
Comment (tristan957) [reply mesonbuild/mdist.py:50]: Complete accident. I'll fix it in a next push if there is a promise that this feature will be merged.

Comment (eli-schwartz) [reply mesonbuild/mdist.py:50]: Yes.

Comment (tristan957) [reply mesonbuild/mdist.py:50]: Eli, last time we talked about this, you and @jpakkane didn't seem to think there was value in bz2 support.

Comment (tristan957) [reply mesonbuild/mdist.py]: Ok. I share the same opinion, basically.

**Comment by tristan957** [comment]
CI is good to go

**Pull request merged**